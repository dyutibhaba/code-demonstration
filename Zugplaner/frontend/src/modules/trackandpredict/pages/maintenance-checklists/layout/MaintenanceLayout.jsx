import React from 'react';
import PropTypes from 'prop-types';

import MainLayout from '../../../../common/components/layout/main-layout/MainLayout';
import Navigation, { breadCrumbShape, linkShape } from '../../../../admin/pages/Navigation';
import { anyNumberOfChildren } from '../../../../common/models/miscellaneousProps';

const MaintenanceLayout = ({
  children,
  links,
  breadcrumb,
  index,
}) => (
  <MainLayout moduleName="Track&Predict" moduleHome="/track-and-predict" breadcrumb={breadcrumb}>
    <Navigation
      links={links}
      selectedIndex={index}
      setTabIndex={() => {}}
    />
    {children}
  </MainLayout>
);

MaintenanceLayout.propTypes = {
  children: anyNumberOfChildren.isRequired,
  links: PropTypes.arrayOf(linkShape).isRequired,
  breadcrumb: PropTypes.arrayOf(breadCrumbShape),
  index: PropTypes.number.isRequired,
};

MaintenanceLayout.defaultProps = {
  breadcrumb: null,
};

export default MaintenanceLayout;
