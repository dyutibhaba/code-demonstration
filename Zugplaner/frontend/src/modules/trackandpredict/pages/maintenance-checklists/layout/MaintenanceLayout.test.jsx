import React from 'react';
import renderWithState from '../../../../../testutils';
import MaintenanceLayout from './MaintenanceLayout';

describe('Maintenance Layout', () => {
  it('renders the maintenance checklist layout correctly', () => {
    expect(renderWithState()((
      <MaintenanceLayout
        links={[{
          to: '/link/to',
          value: 'maintenance.tabs.checklists',
        }]}
        index={0}
      >
        Children
      </MaintenanceLayout>
    ))).toMatchSnapshot();
  });
});
