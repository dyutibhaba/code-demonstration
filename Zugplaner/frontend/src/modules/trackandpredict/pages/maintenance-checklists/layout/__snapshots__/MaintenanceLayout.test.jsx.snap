// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintenance Layout renders the maintenance checklist layout correctly 1`] = `
Array [
  <div
    className="sc-bBXqnf cKDpGo"
  >
    <div
      className="sc-fodVxV BISWi"
    >
      <div
        className="left"
      >
        SideNavigation
        <img
          alt="Protostellar globe"
          className="sc-kEjbxe qRbSs"
          src="ProtostellarGlobe.svg"
        />
        <span
          className="sc-iqHYGH fOGDer"
        >
          <a
            className="sc-bqyKva hDtbLJ"
            href="/track-and-predict"
            onClick={[Function]}
          >
            <span
              className="link"
            >
              Track&Predict
            </span>
          </a>
        </span>
      </div>
      <nav
        className="sc-fFubgz deccwI"
      >
        <div
          className="sc-kstrdz lolpkk tpz-header white"
        >
          <a
            className="sc-hBEYos iRgagf"
            onClick={[Function]}
          >
            navbar.documentation
          </a>
          <a
            className="sc-hBEYos iRgagf"
            href="mailto:support-1logistics@thalesdigital.io"
          >
            <i
              className="material-icons"
            >
              help_outline
            </i>
          </a>
          <a
            className="sc-hBEYos iRgagf"
          >
            <div
              className="tpz-menu-options tpz-header-mobile-menu-options"
            >
              <div
                className="tpz-menu"
              >
                <button
                  className="tpz-menu-button"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <i
                    className="material-icons"
                  >
                    account_circle
                  </i>
                </button>
              </div>
            </div>
          </a>
        </div>
      </nav>
    </div>
  </div>,
  <div
    className="sc-iwyYcG kkPWSB"
  >
    <div
      className="sc-cxFLnm eaKehO"
    >
      <nav
        className="sc-pFZIQ iOsina"
      >
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP  active-nav-item  "
          href="/link/to"
          onClick={[Function]}
        >
          maintenance.tabs.checklists
        </a>
      </nav>
      Children
    </div>
    <div
      className="sc-lmoMRL ktQhwL"
    >
      <div
        className="sc-iBPRYJ ulSNt"
      >
        <div
          className="sc-bdfBwQ bQlKor"
        >
          <div
            className="sc-gsTCUz gSBHTq"
          >
            <div
              className="sc-dlfnbm jFixhj"
            >
              <img
                alt="protostellar-logo"
                height="32px"
                src="Protostellar_Logo_grau_2-02.svg"
              />
            </div>
            <div
              className="sc-hKgILt edhEfs"
            >
              <ul
                className="sc-eCssSg cUiBZU"
              >
                <li
                  className="sc-jSgupP fsVOKa"
                >
                  <a
                    className="sc-gKsewC hFWEic"
                    onClick={[Function]}
                  >
                    footer.notice
                  </a>
                </li>
                <li
                  className="sc-jSgupP fsVOKa"
                >
                  Protostellar
                   
                  0.0.1
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
